msal/src/test/java/com/microsoft/identity/client/e2e/tests/network/nativeauth/GetAccessTokenTests.kt

```diff
@@ -394,47 +436,67 @@ class GetAccessTokenTests : NativeAuthPublicClientApplicationAbstractTest() {
+        Assert.assertTrue(authResult2.scope.contains(scopeA1))
         val tokenWithEmployeeScope2 = authResult2.accessToken
         Assert.assertNotEquals(tokenWithEmployeeScope, tokenWithEmployeeScope2) // New token received
 
         // Reset
         wasRenewAccessTokenInvoked = false
 
-        // Token should be retrieved from API, as the scope belongs to a different API resource
-        val getAccessTokenResult3 = accountState.getAccessToken(forceRefresh = false, scopes = listOf(CUSTOMERS_READ_ALL_SCOPE))
+        // Token should be retrieved from API, as the scope belongs to a different API resource and wasn't requested before
+        val getAccessTokenResult3 = accountState.getAccessToken(forceRefresh = false, scopes = listOf(scopeB1))
         assertState<GetAccessTokenResult.Complete>(getAccessTokenResult3)
         Assert.assertTrue(wasRenewAccessTokenInvoked)
         val authResult3 = (getAccessTokenResult3 as GetAccessTokenResult.Complete).resultValue
-        Assert.assertTrue(authResult3.scope.contains(CUSTOMERS_READ_ALL_SCOPE))
+        Assert.assertTrue(authResult3.scope.contains(scopeB1))
         val tokenWithCustomerScope = authResult3.accessToken
         Assert.assertNotEquals(tokenWithEmployeeScope, tokenWithCustomerScope)
 
         // Reset
         wasRenewAccessTokenInvoked = false
 
         // Token should be retrieved from API, due to force_refresh
-        val getAccessTokenResult4 = accountState.getAccessToken(forceRefresh = true, scopes = listOf(CUSTOMERS_READ_ALL_SCOPE))
+        val getAccessTokenResult4 = accountState.getAccessToken(forceRefresh = true, scopes = listOf(scopeB1))
         assertState<GetAccessTokenResult.Complete>(getAccessTokenResult4)
         Assert.assertTrue(wasRenewAccessTokenInvoked)
         val authResult4 = (getAccessTokenResult4 as GetAccessTokenResult.Complete).resultValue
-        Assert.assertTrue(authResult4.scope.contains(CUSTOMERS_READ_ALL_SCOPE))
+        Assert.assertTrue(authResult4.scope.contains(scopeB1))
         val tokenWithCustomerScope2 = authResult4.accessToken
         Assert.assertNotEquals(tokenWithCustomerScope, tokenWithCustomerScope2) // New token received
     }
+
+    /**
+     * The id token is correctly updated after the access token is refreshed
+     */
+    @Test
+    fun testGetAccessTokenWithForceRefreshThenUpdateIdToken() = runTest {
+        val username = config.email
+        val password = getSafePassword()
+        val result = application.signIn(
+            username = username,
+            password = password.toCharArray()
+        )
+        assertState<SignInResult.Complete>(result)
+        val accountState = (result as SignInResult.Complete).resultValue
+        val idToken1 = accountState.getIdToken()
+        accountState.getAccessToken(forceRefresh = true)
+        val idToken2 = accountState.getIdToken()
+        Assert.assertFalse(idToken1 == idToken2)
```

Comment:
@Yuki-YuXin please use `assertNotEquals`. Besides slightly different validation, it gives more helpful information in case the assert fails.